orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/UpsertDeliveryConfigTask.java

```diff
@@ -0,0 +1,86 @@
+package com.netflix.spinnaker.orca.front50.tasks;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.DeliveryConfig;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class UpsertDeliveryConfigTask implements RetryableTask {
+
+  private Logger log = LoggerFactory.getLogger(getClass());
+
+  private Front50Service front50Service;
+  private ObjectMapper objectMapper;
+  private DeliveryConfigUtils deliveryConfigUtils;
+
+  @Autowired
+  public UpsertDeliveryConfigTask(Front50Service front50Service, ObjectMapper objectMapper) {
+    this.front50Service = front50Service;
+    this.objectMapper = objectMapper;
+    deliveryConfigUtils = new DeliveryConfigUtils(front50Service, objectMapper);
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
```

Comment:
This has been discussed in a few tickets. Not suggesting we solve it here, just reminding/bringing up.
There is a issue with this style upsert in that "last write wins".
If we have two services get the config, tweak it, and upsert it one of them will lose. (obviously, this is an issue currently with a several of endpoints)

Reply:
There's probably a better story here with how to merge updates so you can just change one field, you don't overwrite other people, etc. However for simplicity here I just want the last write to win.